### PR TITLE
Various pre v233 fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/boot/bootctl.c [hunk 1]
```diff
@@ -913,20 +913,31 @@ static int remove_variables(sd_id128_t uuid, const char *path, bool in_order) {
...
         r = sd_id128_get_machine(&machine_id);
         if (r < 0)
                 return log_error_errno(r, "Failed to get machine did: %m");
 
         p = strjoina(esp_path, "/loader/loader.conf");
-        f = fopen(p, "wxe");
-        if (!f)
-                return log_error_errno(errno, "Failed to open loader.conf for writing: %m");
+
+        if (access(p, F_OK) >= 0) /* Silenlty skip creation if the file already exists (early check) */
```
Comment (keszybz) [inline src/boot/bootctl.c]: Typo in "silently".

File: man/systemd.exec.xml [hunk 2]
```diff
@@ -1152,15 +1152,16 @@
...
-        is implied. Note that this option does not prevent kernel tuning through IPC interfaces
-        and external programs. However <varname>InaccessiblePaths=</varname> can be used to
-        make some IPC file system objects inaccessible.</para></listitem>
+        boot-time, with the
+        <citerefentry><refentrytitle>sysctl.d</refentrytitle><manvolnum>5</manvolnum></citerefentry> mechanism. Almost
+        no services need to write to these at runtime; it is hence recommended to turn this on for most services. For
+        this setting the same restrictions regarding mount propagation and privileges apply as for
+        <varname>ReadOnlyPaths=</varname> and related calls, see above. Defaults to off.  If turned on and if running
+        in user mode, or in system mode, but without the <constant>CAP_SYS_ADMIN</constant> capability (e.g. setting
+        <varname>User=</varname>), <varname>NoNewPrivileges=yes</varname> is implied. Note that this option does not
```
Comment (keszybz) [inline man/systemd.exec.xml]: Not added in your patch, but "e.g. setting User=" is very awkward.

Hm, what does it mean "external programs"?

File: man/systemd.exec.xml [hunk 3]
```diff
@@ -1196,7 +1197,8 @@
         unit. Except for container managers no services should require write access to the control groups hierarchies;
         it is hence recommended to turn this on for most services. For this setting the same restrictions regarding
         mount propagation and privileges apply as for <varname>ReadOnlyPaths=</varname> and related calls, see
-        above. Defaults to off.</para></listitem>
+        above. Defaults to off. If <varname>ProtectControlGroups=</varname> is set, <varname>MountAPIVFS=</varname> is
```
Comment (keszybz) [inline man/systemd.exec.xml]: "MountAPIVFS=yes is implied."

File: src/boot/bootctl.c [hunk 4]
```diff
@@ -517,90 +511,59 @@ static int version_check(int fd, const char *from, const char *to) {
...
+                return log_oom();
 
-        r = fflush_and_check(g);
-        if (r < 0) {
-                log_error_errno(r, "Failed to write \"%s\": %m", to);
-                goto error;
+        RUN_WITH_UMASK(0000) {
+                fd_to = open(t, O_WRONLY|O_CREAT|O_CLOEXEC|O_EXCL|O_NOFOLLOW, 0644);
+                if (fd_to < 0)
+                        return log_error_errno(errno, "Failed to create %s: %m", t);
```
Comment (keszybz) [inline src/boot/bootctl.c]: `\"%s\"`, here and elsewhere. I think it helps a lot if the string has a space my mistake or some other strange characters.

**Comment by poettering** [comment]
Force pushed a new version now with the issues @keszybz pointed out addressed

**Pull request merged**